manifests/install.pp

```diff
@@ -4,9 +4,27 @@
 class nrpe::install
 {
   if $nrpe::manage_package {
-    package { $nrpe::package_name:
-      ensure   => installed,
-      provider => $nrpe::provider,
-    }
+    ensure_packages($nrpe::package_name,
```

Comment:
I would prefer to manage those packages with a "normal" `package` resource as this is the main module for this. If you have some special modules which might do the same, I think they should use a `ensure_packages`.

Reply:
So, I can't change it the way you said without a bigger architectural difference, which I fear might make this a harder transition, and this was my compromise.

Someone upgrading to this patch who has set `package_name` expects their list to work.  With this patch, they'd be 'double included', and it'd just work.  And I could set options for myself and be happy.

To use "normal" `package`, we'd have to split all package requirements up so there's no overlap, because otherwise `package` will complain when we include packages during `if $nrpe::manage_package` and then for `if $nrpe::manage_package_daemon`.

To do otherwise, we'd have to flip the defaults to false for my patch, which I could be fine with.  Basically make it an exclusive-or situation. (`$nrpe::manage_package` XOR (if `$nrpe::manage_package_daemon` AND `if $nrpe::manage_package_plugins`)).  I could do that, it'd just mean reversing the defaults on the manage flags.

The other options I see would be major-version-breaking:
- do `if $nrpe::manage_package or $nrpe::manage_package_daemon { package {$nrpe::package_name_daemon} } ` - this would mean `package_name` never gets used, breaking expectations
- drop `if $nrpe::manage_package` completely in favor of my method.

Basically, I agree, using this module, 'it' should control the package (if you let it), which would mean it should be `package`.  I can do flag reversals.  If it's okay to make this a major version change instead of a first step towards deprecation, I can patch it, drop `ensure_packages` back to `package`, and there we go, but that's a hard cut.  Basically, I'm open for whichever seems most sane here.  Just saying, I didn't go for `ensure_packages` willy-nilly. :)

Reply:
Hey @gcoxmoz, thanks for you answer! :) How about doing the following (and `manage_package`/`package_name`)  in `install.pp` based on the values from `init.pp`:

```puppet
  if $nrpe_packages_daemon and $nrpe_packages_plugins {
    $nrpe_packages = [ $nrpe_packages_daemon ] + $nrpe_packages_plugins
  }
```

This would then allow us to only do 1 package resource definition:

```puppet
package { $all_package_which_should_be_installed:
  ensure   => installed,
  provider => $nrpe::provider,
}
```